interceptors/logging/interceptors_test.go

```diff
@@ -110,11 +110,16 @@ func (l *baseMockLogger) Lines() []LogLine {
 
 type mockLogger struct {
 	*baseMockLogger
-
+	m      sync.Mutex
```

Comment:
Do we need a new mutex? Can't we use the same one we're embedding with `baseMockLogger`?

Reply:
Having read this a bit more closely, I think we should just merge `baseMockLogger` and `mockLogger`. I don't see what having two different structs here gives us, and it's really confusing to read. If we have a single `mockLogger` with a mutex that should make it easy to reason about where the locks need to be.

What do you think?

Reply:
Hmm, I guess, we should merge both the `baseMockLogger` and `mockLogger`, which doesn't make sense to break down the logger into two parts.

Regarding the locks, I read that append operations and map operations are not concurrent safe, so I was using it as a rule of thumb for using the locks there. However, if you have more suggestions about the approach, I would be happy to discuss!

Reply:
I'm confused, I looked at the PR again and it doesn't look like you've made the change to merge these two loggers. Do you agree that this is a good idea? If so, could you please try to do it in this PR?

Reply:
I kind of clicked on the review option by mistake :stuck_out_tongue: I am currently working on merging the loggers, will ping you in slack once done.

Reply:
I think we should keep the current structure, and change the name of `baseMockLogger` to `sharedResults` or `sharedLogLines` : More Info - 
https://github.com/grpc-ecosystem/go-grpc-middleware/pull/354#issuecomment-708153888